gui/sharing/forms.py

```diff
@@ -518,11 +518,14 @@ def cleanformset_nfs_share_path(self, formset, forms):
         for network in networks:
             networkobj = IPNetwork(network)
             for unetwork, ustdev in used_networks:
-                try:
-                    unetworkobj = IPNetwork(unetwork)
-                except Exception:
-                    # If for some reason other values in db are not valid networks
-                    unetworkobj = IPNetwork('0.0.0.0/0')
+                if share.nfs_network:
```

Comment:
when we get to this loop, IPNetwork(unetwork) returns an exception so the network portion gets set to 0.0.0.0/0.

This is a problem if you're trying to add another share to the same dataset and enter a specific host IP address.

The idea is that if there is an existing share with a network entry, and you try to add another share with an IP address, this code will set the network portion to 0.0.0.0/0 no matter what and it will error out at line: 529-531.

Reply:
Thats because networks is being set to `['0.0.0.0/0']` in line 490.

Perhaps instead, on line 489 `if not networks:` should be changed to: `if not networks and not self.cleaned_data.get("nfs_hosts"):` ?

Reply:
You're absolutely right. It should be corrected there, instead.

Reply:
My last change causes a traceback in the ipaddr.py library because it's passing an empty string. More investigation is needed.